rewrite the docs

nslater opened the pull request.

chaudum commented on docs/compatibility.rst:
```diff
@@ -0,0 +1,103 @@
+
+Unfortunately, when defining CrateDB `object`_ or `array`_ type columns, you
+must construct the DBAL column definition object manually, using the correct
+class names and attributes.
```
in general, yes.

chaudum commented on docs/compatibility.rst:
```diff
@@ -0,0 +1,98 @@
+------------------
+
+Unfortunately, when defining CrateDB `object`_ or `array`_ type columns, you
+must construct the DBAL column definition object manually, using the correct
```
manually -> programmatically

chaudum replied:
~~object~~

chaudum replied:
using the custom types and attributed provided by the CrateDB DBAL driver.

chaudum commented on docs/compatibility.rst:
```diff
@@ -0,0 +1,98 @@
+Column Definitions
+------------------
+
+Unfortunately, when defining CrateDB `object`_ or `array`_ type columns, you
```
Primitive column types, such as string, integer, ... can be defined in the regular Doctrine way.

nslater commented:
@chaudum thanks for the review! I have addressed all your comments!

there's one final comment from me at the top. do you think you could take a look at it and let me know what you think? ty ty!

nslater commented on docs/appendices/compatibility.rst:
```diff
@@ -0,0 +1,78 @@
+|                |                                       |            |
+|                | - Joins                               |            |
+|                | - Fulltext indexes                    |            |
++----------------+---------------------------------------+------------+
```
@chaudum @Taliik can you verify the table above?

nslater replied:
verified and fixed. thanks

nslater commented:
@chaudum @Taliik I have pushed some fixups. can I get another review please? ty!

nslater commented on docs/appendices/data-types.rst:
```diff
@@ -15,23 +15,24 @@ This driver maps CrateDB types to the following PHP types:
+   "`geo_point`__", "`array`__"
+   "`geo_shape`__", "`object`__"
    "`object`__", "`MapType`__"
    "`array`__", "`ArrayType`__"
```
@seut can you verify I've done this right, per your comments on trello?

nslater commented:
@seut can I get a final review on this to merge it?

last remaining comment is here:

https://github.com/crate/crate-dbal/pull/48#discussion_r186176720

chaudum commented on .gitignore:
```diff
@@ -1,5 +1,6 @@
 .idea
 .installed.cfg
+.installed.cfg
```
Is this a duplicated that slipped in?

seut commented on docs/appendices/data-types.rst:
```diff
@@ -0,0 +1,112 @@
+   "`ip`__", "`string`__"
+   "`timestamp`__", "`TimestampType`__"
+   "`geo_point`__", "`array`__"
+   "`geo_shape`__", "`object`__"
```
php type is `array`

seut commented on docs/appendices/data-types.rst:
```diff
@@ -0,0 +1,112 @@
+   "`timestamp`__", "`TimestampType`__"
+   "`geo_point`__", "`array`__"
+   "`geo_shape`__", "`object`__"
+   "`object`__", "`MapType`__"
```
php type is `array`

seut commented on docs/appendices/data-types.rst:
```diff
@@ -0,0 +1,112 @@
+   "`geo_point`__", "`array`__"
+   "`geo_shape`__", "`object`__"
+   "`object`__", "`MapType`__"
+   "`array`__", "`ArrayType`__"
```
php type is `array`

seut commented on docs/appendices/data-types.rst:
```diff
@@ -0,0 +1,112 @@
+   "`double`__", "`float`__"
+   "`string`__", "`string`__"
+   "`ip`__", "`string`__"
+   "`timestamp`__", "`TimestampType`__"
```
php type is `DateTime`

nslater commented:
@seut okay pushed a fixup! ty!

nslater commented:
@seut pushed another fixup thanks!
